src/aspire/source/image.py

```diff
@@ -27,7 +29,50 @@
 logger = logging.getLogger(__name__)
 
 
-class ImageSource:
+class ImageAccessor:
+    """
+    Helper class for accessing images from an ImageSource as slices via the `src.images[start:stop:step]` API.
+    """
+
+    def __init__(self, fun, num_imgs):
+        """
+        :param fun: The private image-accessing method specific to the ImageSource associated with this ImageAccessor.
+                    Generally _images() but can be substituted with a custom method.
+        :param num_imgs: The max number of images that this ImageAccessor can load (generally ImageSource.n).
+        """
+        self.fun = fun
+        self.num_imgs = num_imgs
+
+    def __getitem__(self, indices):
+        """
+        ImageAccessor can be indexed via Python slice object, 1-D NumPy array, list, or a single integer,
+        corresponding to the indices of the requested images. By default, slices default to a start of 0,
+        an end of self.num_imgs, and a step of 1.
+        :return: An Image object containing the requested images.
+        """
+        if isinstance(indices, Iterable) and not isinstance(indices, np.ndarray):
+            indices = np.fromiter(indices, int)
+        if isinstance(indices, (int, np.integer)):
```

Comment:
elif

Not for performance or anything, it just helps document intent in the code. (In case someone else needs to refactor it later).

Reply:
done